src/ensembl/src/content/app/entity-viewer/shared/helpers/entity-helpers.ts

```diff
@@ -152,12 +158,32 @@ export const getLongestProteinLength = (gene: GetLongestProteinLengthParam) => {
 
 export enum ExternalSource {
   INTERPRO = 'Interpro',
-  UNIPROT = 'UniProt',
+  UNIPROT_TREMBL = 'UniProtKB/TrEMBL',
+  UNIPROT_SWISSPROT = 'UniProtKB/Swiss-Prot',
   PDBE = 'PDBe-KB'
 }
 
 export const externalSourceLinks = {
   [ExternalSource.INTERPRO]: 'https://www.ebi.ac.uk/interpro/protein/UniProt/',
-  [ExternalSource.UNIPROT]: 'https://www.uniprot.org/uniprot/',
+  [ExternalSource.UNIPROT_TREMBL]: 'https://www.uniprot.org/uniprot/',
+  [ExternalSource.UNIPROT_SWISSPROT]: 'https://www.uniprot.org/uniprot/',
   [ExternalSource.PDBE]: 'https://www.ebi.ac.uk/pdbe/pdbe-kb/proteins/'
 };
+
+export const getProteinXrefs = (transcript: {
+  product_generating_contexts: Array<{
+    product: Pick<ProductWithoutDomains, 'external_references'>;
+  }>;
+}) => {
+  const xrefs =
+    transcript.product_generating_contexts[0].product.external_references;
+  let proteinXrefs = xrefs.filter(
+    (xref) => xref.source.id === SWISSPROT_SOURCE
+  );
+
+  if (!proteinXrefs.length) {
+    proteinXrefs = xrefs.filter((xref) => xref.source.id === SPTREMBL_SOURCE);
+  }
+
+  return proteinXrefs;
+};
```

Comment:
I've played with the types some more. Here's the type I suggest to use for this function:

```ts
export const getProteinXrefs = <T extends Pick2<ExternalReference, 'source', 'id'>>(transcript: {
  product_generating_contexts: Array<{
    product: {
      external_references: T[];
    }
  }>;
}) => {
  const xrefs =
    transcript.product_generating_contexts[0].product.external_references;
  let proteinXrefs = xrefs.filter(
    (xref) => xref.source.id === SWISSPROT_SOURCE
  );

  if (!proteinXrefs.length) {
    proteinXrefs = xrefs.filter((xref) => xref.source.id === SPTREMBL_SOURCE);
  }

  return proteinXrefs;
};
```

**Reasoning:**
- You will notice that this function only cares about the `xref.source.id` field, whereas the `ProductWithoutDomains` has `external_references` that contain other fields that are irrelevant for this function.
- Using the generic type `T extends Pick2<ExternalReference, 'source', 'id'>` allows us to pass to this function external references that have any set of fields as long as one of those fields is `source` with an `id` — and to make typescript understand that whatever external_references shape was passed to this function will be exactly the same shape as the function is going to return.